tests/system/feature_store/test_spark_engine.py

```diff
@@ -1422,6 +1423,72 @@ def test_ingest_with_steps_mapval(self, with_original_features):
         csv_path_storey = measurements.get_target_path(name="csv")
         self.read_csv_and_assert(csv_path_spark, csv_path_storey)
 
+    def test_mapval_with_partial_mapping(self):
+        # we need this test in order to check partially_mapping -> only part of the values in field are replaced.
+        key = "patient_id"
+        csv_path_spark = "v3io:///bigdata/test_mapvalues_with_partial_mapping"
+
+        measurements = fstore.FeatureSet(
+            "measurements_spark",
+            entities=[fstore.Entity(key)],
+            timestamp_key="timestamp",
+            engine="spark",
+        )
+        measurements.graph.to(
+            MapValues(
+                mapping={
+                    "bad": {17: 0},
+                },
+                with_original_features=True,
+            )
+        )
+        source = ParquetSource("myparquet", path=self.get_remote_pq_source_path())
+        targets = [CSVTarget(name="csv", path=csv_path_spark)]
+        fstore.ingest(
+            measurements,
+            source,
+            targets,
+            spark_context=self.spark_service,
+            run_config=fstore.RunConfig(local=False),
+        )
+        csv_path_spark = measurements.get_target_path(name="csv")
+        df = self.read_csv(csv_path=csv_path_spark)
+        assert not df["bad_mapped"].isna().any()
+        assert not df["bad_mapped"].isnull().any()
+        assert not (df["bad_mapped"] == 17).any()
```

Comment:
Might want to asset more than just that. The file could contain no data and still all asserts would pass.